pkg/common/config/config.go

```diff
@@ -456,3 +475,14 @@ func GetSupervisorNamespace(ctx context.Context) (string, error) {
 	}
 	return string(namespace), nil
 }
+
+// GetClusterFlavor returns the cluster flavor based on the env variable set in the driver deployment file
+func GetClusterFlavor(ctx context.Context) cnstypes.CnsClusterFlavor {
+	// CLUSTER_FLAVOR is defined only in Supervisor and Guest cluster deployments.
+	// If it is empty, it is implied that cluster flavor is Vanilla K8S
+	clusterFlavor := cnstypes.CnsClusterFlavor(os.Getenv("CLUSTER_FLAVOR"))
```

Comment:
I think we should take the opportunity to validate permitted set of flavors and return error if any other random string is specified as `CLUSTER_FLAVOR`.

Reply:
Done